### PR TITLE
Style sheets

### LINKED ISSUE #81: Support css styling with style-tags
Both within the svg as well as somewhere else in the HTML DOM.

### PR DESCRIPTION
**Pull request by zenker981**
Closes #81

### CODE REVIEW

File: src/css-select-browser-adapter-master/.gitattributes [hunk 1]
```diff
@@ -0,0 +1,17 @@
+# Auto detect text files and perform LF normalization
```
Comment (HackbrettXXX) [inline src/css-select-browser-adapter-master/.gitattributes]: Don't include the browser adapter in our repo. It is also available on npm:
https://www.npmjs.com/package/css-select-browser-adapter

File: src/svg2pdf.js [hunk 2]
```diff
@@ -151,16 +153,103 @@ SOFTWARE.
...
+  }
+
+  StyleSheets.prototype.getCompiledRule = function(selector) {
+    return this.selectorMap[selector] || (this.selectorMap[selector] = CSSselect.compile(selector, {adapter: BrowserDomAdapter}));      
+  }
+
   // returns an attribute of a node, either from the node directly or from css
-  var getAttribute = function (node, propertyNode, propertyCss) {
-    propertyCss = propertyCss || propertyNode;
+  var getAttribute = function (node, propertyNames, styleSheets) {
```
Comment (HackbrettXXX) [inline src/svg2pdf.js]: we could define getAttribute on the context object, as we need a context anyways for this.

File: src/svg2pdf.js [hunk 3]
```diff
@@ -151,16 +153,103 @@ SOFTWARE.
...
+      for (var text of sheetTexts) {
+        var sheet = Css.parse(text, {silent: true});
+        this.sheets.push(sheet);
+      }
+    }
+    return this.sheets;
+  }
+
+  StyleSheets.loadSheet = function (url) {
+    const request = new XMLHttpRequest();
```
Comment (HackbrettXXX) [inline src/svg2pdf.js]: We could try and use the [styleSheets API](https://developer.mozilla.org/en-US/docs/Web/API/DocumentOrShadowRoot/styleSheets) for this. Don't know if this works, though.

Comment (zenker981) [reply src/svg2pdf.js]: I retried it and couldn't find any styleSheets in the debug.html DOM...

File: src/svg2pdf.js (hunk 2 above)
Comment (zenker981) [reply src/svg2pdf.js]: Actually I tried to implement it in a way that you can use getAttribute() without any context...

**Pull request merged**